core/app/models/spree/zone.rb

```diff
@@ -58,6 +58,8 @@ def self.match(address)
     # current zone, if it's a state zone. If the passed-in zone has members, it
     # will also be in the result set.
     def self.with_shared_members(zone)
+      return none unless zone
```

Comment:
Looks like it should to me. This is a class method so it will return the `Spree::Zone.none` null object relation.